src/data_analysis/symmetry_checker.py

```diff
@@ -0,0 +1,239 @@
+    it estimates (per-patient) the Pearson correlation coefficient between
+    the left and right signals, and creates Bland-Altman plots for a graphical
+    analysis.
+    """
+
+    def __init__(self, data_dir, out_dir, columns, resample=None):
+        """
+        Arguments:
+            data_dir: Input folder with the .csv files. The following naming
+                      convention applies for the .csv files:
+                            ([0-9]*)(L|R).*
+                      The first capture group identifies the patient id, the
+                      second one whether the data is from left or right side.
+                            001L_storage-vital.csv
+                            001R_storage-vital.csv
+            out_dir: Path to output directory
+            columns: Columns to select ("columns of interest", coi)
+            resample: Optionally resample the data. For instance, setting
+                      resample="30s" aggregates the data into 30s bins.
+                      See doc of pd.DataFrame.resample for details.
+
+        """
+        self._loader = PatientDataLoader()
+        self._data_dir = Path(data_dir)
+        self._out_dir = Path(out_dir)
+        self._resample = resample
+        # Columns of interest.
+        self._coi = columns
+
+    def _load_data(self, file_left, file_right):
+        def cols_filter(x):
+            # Ensure only columns with dtype float.
+            return x not in ["de_morton_label"]
+        from collections import defaultdict
+        dtypes = defaultdict(lambda: float)
+        dtypes["timestamp"] = str
+        df_left = self._loader.load_everion_patient_data(dir_name=self._data_dir,
+                                                         filename=file_left.name,
+                                                         csv_delimiter=';',
+                                                         usecols=cols_filter,
+                                                         dtype=dtypes)
+        df_right = self._loader.load_everion_patient_data(dir_name=self._data_dir,
+                                                          filename=file_right.name,
+                                                          csv_delimiter=';',
+                                                          usecols=cols_filter,
+                                                          dtype=dtypes)
+        df_left = df_left.set_index("timestamp")
+        df_right = df_right.set_index("timestamp")
+        # select columns of interest
+        df_left = df_left[self._coi+["de_morton"]]
+        df_right = df_right[self._coi+["de_morton"]]
+        # trick to prepend column level
+        df_left = pd.concat([df_left], keys=["left"], axis=1)
+        df_right = pd.concat([df_right], keys=["right"], axis=1)
+        df_left = df_left.reorder_levels([1,0], axis=1)
+        df_right = df_right.reorder_levels([1,0], axis=1)
+        # join on index
```

Comment:
Imho these comments are not needed since the code is self-explanatory

Reply:
I agree!